src/Simplex/Chat/Messages.hs

```diff
@@ -1178,14 +1178,51 @@ data RcvMessage = RcvMessage
 
 type MessageId = Int64
 
--- TODO [channels fwd] review
-data ForwardingJob = ForwardingJob
-  { jobId :: Int64,
-    msgBatchEncoding :: Maybe (NonEmpty (ChatMessage 'Json)),
-    cursorGroupMemberId :: Maybe Int64
+-- TODO [channels fwd] review types
+data ForwardTask = ForwardTask
+  { taskId :: Int64,
+    taskContext :: ForwardTaskContext
   }
   deriving (Show)
 
+data ForwardTaskContext
+  = FTCMessage {messageForwardContext :: MessageForwardContext}
+  | FTCRelayRemoval {relayRemovalForwardContext :: RelayRemovalForwardContext}
+  | FTCReactionCount {reactionCountForwradContext :: ReactionCountForwradContext}
+  deriving (Show)
+
+-- prevSenderInteractionTs: members are split based on this timestamp;
+-- postInteractionCursorGMId: for these members sender profiles are packaged;
+-- question: how to retrieve sender profiles for multiple messages when we already created encoding/list of them?
+-- perhaps we should save references to messages on the task, instead of encoding;
+data MessageForwardContext = MessageForwardContext
+  { prevSenderInteractionTs :: Maybe UTCTime,
+    postInteractionCursorGMId :: Maybe Int64,
+    preInteractionCursorGMId :: Maybe Int64,
+    messages :: Maybe (NonEmpty (ChatMessage 'Json)),
+  }
+  deriving (Show)
+
+-- messageId: we don't know is it group deletion or relay removal;
+-- we could read ChatMessage 'Json by reference instead;
+data RelayRemovalForwardContext = RelayRemovalForwardContext
+  { cursorGMId :: Maybe Int64,
+    messageId :: MessageId
+  }
+  deriving (Show)
+
+-- reaction counts to be retrieved from chat items;
+-- question: how to keep track which chat items received reactions since last such task?
+-- requires additional protocol message
+data ReactionCountForwradContext = ReactionCountForwradContext
```

Comment:
```suggestion
data ChatItemCountsTask = ChatItemCountsTask
```

it would cover comments count, so shouldn't be specialized to reactions.

Also I don't understand why you need separate types and not just fields in constructors?

There are three layers of types when only one seems necessary.